packages/engine/lib/stateful/src/field/mod.rs

```diff
@@ -1,13 +1,23 @@
-//! Provides an hEngine-specific representation of Arrows `Field` and `DataType`.
+//! Provides an hEngine-specific representation of Arrows `Field`, `DataType`, and `Schema`.
 //!
-//! As only a selection of [`arrow::DataType`]s is provided to hEngine, this module defines
-//! [`FieldType`], which uses Arrow types underneath and adds an `Any` type, which internally uses
-//! a JSON encoded `String` type. As [`FieldType`] can't be used in [`arrow::Field`], this module
-//! provides [`FieldSpec`] to associate a name with the [`FieldType`].
+//! As only a selection of Arrow [`DataType`]s is supported at hEngine, this module defines
+//! [`FieldType`], which uses Arrow types underneath and adds an `Any` type, which can represent
+//! arbitrary data. As [`FieldType`] can't be used in Arrows [`Field`], this module provides
+//! [`FieldSpec`] to associate a name with the [`FieldType`].
 //!
-//! For naming a [`FieldSpec`], [`FieldKey`] is provided, as a name has strict requirements on how
-//! it's built. [`FieldKey`] is a unique identifier for a given name in a provided [`FieldScope`]
-//! and its [`FieldSource`].
+//! To create an Arrow [`Schema`], this module provides a [`FieldSpecMap`], which maps a
```

Comment:
Could be good to say why we create a Schema. "We use Arrow to store Fields in memory inside a RecordBatch, we generate Schemas for that batch by using a `FieldSpecMap`..."

Reply:
Addressed this in b1b7b08f4269775b94e6a768bb1db2015ff6030e